dspace-server-webapp/src/main/java/org/dspace/app/rest/ItemUploadController.java

```diff
@@ -0,0 +1,155 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@RestController
+@RequestMapping("/api/core/items/{uuid}")
+public class ItemUploadController {
+
+    private static final Logger log = LogManager.getLogger();
+
+    @Autowired
+    protected Utils utils;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private BitstreamService bitstreamService;
+
+    @Autowired
+    private BitstreamConverter bitstreamConverter;
+
+    @Autowired
+    private MetadataConverter metadataConverter;
+
+    @Autowired
+    private BitstreamFormatService bitstreamFormatService;
+
+    @RequestMapping(method = RequestMethod.POST, value = "/bitstreams", headers = "content-type=multipart/form-data")
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'WRITE') && hasPermission(#uuid, 'ITEM', 'ADD')")
+    public BitstreamResource uploadBitstream(HttpServletRequest request, @PathVariable UUID uuid,
+                                             @RequestParam("file") MultipartFile uploadfile,
+                                             @RequestParam(value = "properties", required = false) String properties) {
+
+        Context context = ContextUtil.obtainContext(request);
+        Item item = null;
+        Bitstream bitstream = null;
+        try {
+            item = itemService.find(context, uuid);
+        } catch (SQLException e) {
+            log.error("Something went wrong trying to find the Item with uuid: " + uuid, e);
+        }
+        if (item == null) {
+            throw new ResourceNotFoundException("The given uuid did not resolve to an Item on the server: " + uuid);
+        }
+        InputStream fileInputStream = null;
+        try {
+            fileInputStream = uploadfile.getInputStream();
+        } catch (IOException e) {
+            log.error("Something went wrong when trying to read the inputstream from the given file in the request");
+            throw new UnprocessableEntityException("The InputStream from the file couldn't be read");
+        }
+        try {
+            bitstream = processBitstreamCreation(context, item, fileInputStream, properties,
+                                                 uploadfile.getOriginalFilename());
+            itemService.update(context, item);
+            context.commit();
+        } catch (AuthorizeException | IOException | SQLException e) {
+            log.error(
+                "Something went wrong with trying to create the single bitstream for file with filename: " + uploadfile
+                    .getOriginalFilename()
+                    + " for item with uuid: " + uuid + " and possible properties: " + properties);
```

Comment:
Again, exception should be logged. Otherwise we don't know what went wrong.

Reply:
This has been updated